### PR TITLE
feat(client): catch errors in services instead of in HttpHelper

### PR DESCRIPTION
**Pull request by guyts96**

### CODE REVIEW

**Review by souziel — APPROVED** [review]

File: client/src/services/roomService.ts [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileDeleting'), 'error');
+            });
     }
 
     static async updateRoom(room: Room): Promise<void> {
-        return HttpHelper.put(`/room/${room.roomId}`, room);
+        return http.put<void, Room>(`/room/${room.roomId}`, room)
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileDeleting'), 'error');
```
Comment (Lukyio) [inline client/src/services/roomService.ts]: Is this update function equivalent to a delete?

File: client/src/services/userService.ts [hunk 2]
```diff
@@ -22,34 +22,63 @@ export default class UserService {
...
+                return '';
+            });
     }
 
     static async isLocked(): Promise<boolean> {
-        return http.get('/user/isLocked');
+        return http.get<boolean>('/user/isLocked')
+            .then((b) => b)
+            .catch(() => {
+                toastService.toast(tr('user.errorWhileLocking'), 'error');
```
Comment (Lukyio) [inline client/src/services/userService.ts:35]: what does isLocked do?

Comment (guyts96) [reply client/src/services/userService.ts:35]: It checks if the front end is locked

Comment (asoran) [reply client/src/services/userService.ts:35]: It checks if the application is locked (verrouiller) or not

File: client/src/services/roomService.ts (hunk 1 above)
Comment (guyts96) [reply client/src/services/roomService.ts]: Nop, that's a typo

**Review by Lukyio — APPROVED** [review]
♌

**Review by asoran — APPROVED** [review]
🥇

**Pull request merged**